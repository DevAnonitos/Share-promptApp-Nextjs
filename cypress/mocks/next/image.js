import { ImageProps } from 'next/image';


/**
 * Converts the next/image static image URL to a regular path.
 *
 * Example:
 *
 * /_next/static/media/404.ea2b1f50.png -> /assets/images/404.png
 */
const convertURL = (url) => {
  return url
    .replace(/\/_next\/static\/media\//, '/assets/images/') // Use actual images location
    .replace(/(?<=\.)(.+)(?=png|jp?eg|tiff?|png|webp|bmp|gif)/, '');
};

const Image = (props) => {
  // Regular path to image resource
  if (typeof props.src === 'string') {
    return <img src={props.src} alt="image element" />;
  }

  let src;
  // StaticImageData - an import of image resource
  if ('src' in props.src) {
    src = props.src.src;
  } else {
    // StaticRequire
    src = props.src.default.src;
  }

  return <img src={convertURL(src)} alt="image element" />;
};

export default Image;