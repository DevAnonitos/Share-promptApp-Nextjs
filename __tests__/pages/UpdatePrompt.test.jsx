import { render, screen } from "@testing-library/react";
import UpdatePrompt from "@/app/update-prompt/page";
import React from "react";
import { useRouter, useSearchParams } from "next/navigation";
import { useSession } from "next-auth/react";


describe("UpdatePrompt", () => {
  it('should render UpdatePrompt Page', () => {
    expect(UpdatePrompt).toBeDefined()
  });
})