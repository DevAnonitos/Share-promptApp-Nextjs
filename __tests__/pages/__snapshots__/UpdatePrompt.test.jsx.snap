// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdatePrompt should render UpdatePrompt Page 1`] = `[Function]`;
